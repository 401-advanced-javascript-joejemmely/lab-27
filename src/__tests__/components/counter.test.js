import React from 'react';
import Counter from '../../components/counter/counter';

describe('Counter', () => {
  it('renders', () => {
    const mountedSimple = shallow(<Counter />);
    expect(mountedSimple.find('.counter')).toBeTruthy();
  });

  it('increment state', () => {
    const mountedSimple = mount(<Counter />);
    const buttonUp = mountedSimple.find('.up.clicker').at(0);

    buttonUp.simulate('click');
    expect(mountedSimple.state('count')).toEqual(1);
  });

  it('decrement state', () => {
    const mountedSimple = mount(<Counter />);
    const buttonDown = mountedSimple.find('.down.clicker').at(0);

    buttonDown.simulate('click');
    expect(mountedSimple.state('count')).toEqual(-1);
  });
});
