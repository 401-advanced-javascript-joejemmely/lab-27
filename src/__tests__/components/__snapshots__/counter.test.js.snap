// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter rendering follows the snapshot 1`] = `
<section
  className="counter css-aiy23l"
>
  <a
    className="down clicker css-kmz3n4"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count  css-13u058o"
  >
    0
  </span>
  <a
    className="up clicker css-1kyj0em"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
